### PR TITLE
Adds a weight bench

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

You can work out and it buffs your cqc by 1 for 15 minutes
Leftclick to bench, rightclick to change weights
Added 1 bench per map for now
Can be moved with wrench
Benching max weights has a 10% chance of breaking your arms

## Why It's Good For The Game

https://tenor.com/bpaOl.gif

## Changelog
:cl: Tivi/Gregor
add: Added a weight bench
/:cl:

### CODE REVIEW

**Comment by Xander3359** [comment]
I am going to bench maxx every round so I can walk into medbay roundstart with broken arms always. Speedmerge this asap.

**Comment by Gregormack** [comment]
Finally! The feature worth stats dumping

**Comment by novaepee** [comment]
> buffs your cqc by 1 for 15 minutes

Vali users hogging bench at round start.

> Can be moved with wrench

Likely a vali user will move bench press to FOB and power min-max this.

**Comment by Losenis** [comment]
Gym tadpole is real

File: code/game/objects/structures/benchpress.dm [hunk 1]
```diff
@@ -0,0 +1,125 @@
+/obj/structure/benchpress
+	name = "weight training bench"
+	desc = "Just looking at this thing makes you feel tired. Leftclick to bench, rightclick to change weights"
```
Comment (XSlayer300) [inline code/game/objects/structures/benchpress.dm:3]:
```suggestion
	desc = "Just looking at this thing makes you feel tired.<br>Left click to bench, right click to change weights."
```

File: code/game/objects/structures/benchpress.dm [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	flick("swap_[plates]", src)
+
+
+
+/obj/structure/benchpress/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	if(.)
+		return
+	if(HAS_TRAIT(src, BENCH_BEING_USED))
+		balloon_alert(user, "wait your turn!")
```
Comment (XSlayer300) [inline code/game/objects/structures/benchpress.dm:61]:
```suggestion
		balloon_alert(user, "wait for your turn!")
```

Comment (Xander3359) [reply code/game/objects/structures/benchpress.dm:61]: less word better, even if grammar bad.
Balloon alert moment.

File: code/game/objects/structures/benchpress.dm [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+		return FALSE
+	tool.play_tool_sound(src)
+	balloon_alert(user, "deconstructing...")
+	if(!do_after(user, 10 SECONDS, target = src))
+		return FALSE
+	new /obj/item/stack/sheet/metal(get_turf(src))
+	new /obj/item/stack/rods(get_turf(src))
+	new /obj/item/stack/rods(get_turf(src))
+	qdel(src)
+	return TRUE
```
Comment (Lumipharon) [inline code/game/objects/structures/benchpress.dm:125]: before new procs?

File: code/game/objects/structures/benchpress.dm [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+	creak_loop.start(src)
+
+///cleans up releases exerciser
+/obj/structure/benchpress/proc/finish_press(mob/user)
+	creak_loop.stop(src)
+	playsound(user, 'sound/machines/click.ogg', 60, TRUE)
+	REMOVE_TRAIT(src, BENCH_BEING_USED, WEIGHTBENCH_TRAIT)
+	user.flags_atom &= ~DIRLOCK
+	REMOVE_TRAIT(user, TRAIT_IMMOBILE, WEIGHTBENCH_TRAIT)
+	if(plates >= 5 && prob(10))
```
Comment (Lumipharon) [inline code/game/objects/structures/benchpress.dm:81]: number of plates making literally no difference except to break your arms is kinda funny.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/benchpress.dm (hunk 4 above)
Comment (Gregormack) [reply code/game/objects/structures/benchpress.dm:81]: should be some progression and benefits. But that wasn't discussed thoroughly

Comment (TiviPlus) [reply code/game/objects/structures/benchpress.dm:81]: Shrug someone can add some later if they think of it

File: code/game/objects/structures/benchpress.dm (hunk 3 above)
Comment (TiviPlus) [reply code/game/objects/structures/benchpress.dm:125]: Done

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**